hearth/lib/hearth/telemetry/telemetry_provider.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module Hearth
+  module Telemetry
+    # Base for all TelemetryProvider classes
+    class TelemetryProvider
+      def initialize(tracer_provider: nil, context_manager: nil)
```

Comment:
I think we may want to document this method since its effectively part of the public api.